src/Formik.tsx

```diff
@@ -445,16 +445,17 @@ export class Formik<Values = FormikValues> extends React.Component<
       this.setState({ isValidating: false });
       const isValid = Object.keys(combinedErrors).length === 0;
       if (isValid) {
-        this.executeSubmit();
+        return Promise.resolve(this.executeSubmit());
```

Comment:
Why wrap this with `Promise.resolve`? Isn't the result the same without it?

Reply:
This guarantees that a promise is returned. If `this.props.onSubmit` were to return `void` instead of a promise, this would ensure chaining.